core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java

```diff
@@ -1,38 +1,115 @@
 package org.stellar.anchor.sep6;
 
 import com.google.common.collect.ImmutableMap;
+import java.time.Instant;
 import java.util.*;
 import java.util.stream.Collectors;
 import org.apache.commons.lang3.NotImplementedException;
+import org.stellar.anchor.api.event.AnchorEvent;
 import org.stellar.anchor.api.exception.*;
 import org.stellar.anchor.api.sep.AssetInfo;
+import org.stellar.anchor.api.sep.SepTransactionStatus;
 import org.stellar.anchor.api.sep.sep6.*;
 import org.stellar.anchor.api.sep.sep6.InfoResponse.*;
 import org.stellar.anchor.api.shared.RefundPayment;
 import org.stellar.anchor.api.shared.Refunds;
 import org.stellar.anchor.asset.AssetService;
 import org.stellar.anchor.auth.Sep10Jwt;
 import org.stellar.anchor.config.Sep6Config;
+import org.stellar.anchor.event.EventService;
+import org.stellar.anchor.util.MemoHelper;
+import org.stellar.anchor.util.SepHelper;
+import org.stellar.anchor.util.TransactionHelper;
+import org.stellar.sdk.KeyPair;
+import org.stellar.sdk.Memo;
 
 public class Sep6Service {
   private final Sep6Config sep6Config;
   private final AssetService assetService;
   private final Sep6TransactionStore txnStore;
+  private final EventService.Session eventSession;
 
   private final InfoResponse infoResponse;
 
   public Sep6Service(
-      Sep6Config sep6Config, AssetService assetService, Sep6TransactionStore txnStore) {
+      Sep6Config sep6Config,
+      AssetService assetService,
+      Sep6TransactionStore txnStore,
+      EventService eventService) {
     this.sep6Config = sep6Config;
     this.assetService = assetService;
     this.txnStore = txnStore;
+    this.eventSession =
+        eventService.createSession(this.getClass().getName(), EventService.EventQueue.TRANSACTION);
     this.infoResponse = buildInfoResponse();
   }
 
   public InfoResponse getInfo() {
     return infoResponse;
   }
 
+  public GetDepositResponse deposit(Sep10Jwt token, GetDepositRequest request)
+      throws AnchorException {
+    // Pre-validation
+    if (token == null) {
+      throw new SepNotAuthorizedException("missing token");
+    }
+    if (request == null) {
+      throw new SepValidationException("missing request");
+    }
+
+    AssetInfo asset = assetService.getAsset(request.getAssetCode());
+    if (asset == null || !asset.getDeposit().getEnabled() || !asset.getSep6Enabled()) {
+      throw new SepValidationException(
+          String.format("invalid operation for asset %s", request.getAssetCode()));
+    }
+
+    try {
+      KeyPair.fromAccountId(request.getAccount());
+    } catch (RuntimeException ex) {
+      throw new SepValidationException(String.format("invalid account %s", request.getAccount()));
+    }
+    Memo memo = MemoHelper.makeMemo(request.getMemo(), request.getMemoType());
+    String id = SepHelper.generateSepTransactionId();
+
+    Sep6TransactionBuilder builder =
+        new Sep6TransactionBuilder(txnStore)
```

Comment:
We store client_domain in sep24, is it not needed to store for sep-6?
(I'm guessing not -> we can get in from SEP-10 token)

Reply:
I think so, but SEP-31 also stores it. Let's add it later if we find we need it.